ts/components/ui/ListItemIDP.tsx

```diff
@@ -0,0 +1,147 @@
+  IOListItemIDPHSpacing,
+  IOListItemIDPSavedVSpacing,
+  IOListItemLogoMargin
+} from "../core/variables/IOSpacing";
+
+type ListItemIDP = WithTestID<{
+  name: string;
+  localLogo: ImageSourcePropType;
+  logo: ImageSourcePropType;
+  saved?: boolean;
+  onPress: (event: GestureResponderEvent) => void;
+}>;
+
+const styles = StyleSheet.create({
+  button: {
+    borderWidth: 1,
+    borderColor: IOColors["grey-100"],
+    borderRadius: IOListItemIDPRadius,
+    backgroundColor: IOColors.white,
+    flexDirection: "row",
+    justifyContent: "space-between",
+    alignItems: "center",
+    paddingVertical: IOListItemIDPVSpacing,
+    paddingHorizontal: IOListItemIDPHSpacing
+  },
+  idpName: {
+    color: IOColors["grey-700"],
+    fontSize: 12,
+    lineHeight: 16,
+    alignSelf: "center",
+    textTransform: "uppercase",
+    flexShrink: 1,
+    ...makeFontStyleObject("Regular", false, "ReadexPro")
+  },
+  idpLogo: {
+    marginStart: IOListItemLogoMargin,
+    width: 120,
+    height: 30,
+    resizeMode: "contain"
+  }
+});
+
+// https://github.com/facebook/react-native/issues/12606
+// Image cache forced refresh for Android by appending
+// the `ts` query parameter as DDMMYYYY to simulate a 24h TTL.
+const androidIdpLogoForcedRefreshed = () =>
+  Platform.OS === "android" ? `?ts=${toAndroidCacheTimestamp()}` : "";
+
+export const ListItemIDP = ({
+  name,
+  localLogo,
+  logo,
+  saved,
+  onPress,
+  testID
+}: ListItemIDP) => {
+  const isPressed: Animated.SharedValue<number> = useSharedValue(0);
+
+  // Scaling transformation applied when the button is pressed
+  const animationScaleValue = IOScaleValues?.magnifiedButton?.pressedState;
+
+  const scaleTraversed = useDerivedValue(() =>
+    withSpring(isPressed.value, IOSpringValues.button)
+  );
+
+  // Interpolate animation values from `isPressed` values
+  const animatedStyle = useAnimatedStyle(() => {
+    const scale = interpolate(
+      scaleTraversed.value,
+      [0, 1],
+      [1, animationScaleValue],
+      Extrapolate.CLAMP
+    );
+
+    return {
+      transform: [{ scale }]
+    };
+  });
+
+  const onPressIn = useCallback(() => {
+    // eslint-disable-next-line functional/immutable-data
+    isPressed.value = 1;
+  }, [isPressed]);
+  const onPressOut = useCallback(() => {
+    // eslint-disable-next-line functional/immutable-data
+    isPressed.value = 0;
+  }, [isPressed]);
+
+  const urlLogoIDP = localLogo
+    ? localLogo
+    : {
+        uri: `${logo}${androidIdpLogoForcedRefreshed()}`
+      };
+
+  return (
+    <Pressable
+      onPress={onPress}
+      onPressIn={onPressIn}
+      onPressOut={onPressOut}
+      onTouchEnd={onPressOut}
+      accessible={true}
+      accessibilityRole={"button"}
+      accessibilityLabel={name}
+      testID={testID}
+    >
+      <Animated.View
+        style={[
+          styles.button,
+          saved && { paddingVertical: IOListItemIDPSavedVSpacing },
+          animatedStyle
+        ]}
+      >
```

Comment:
As already mentioned by @forrest57 in a recent PR, since the Pressed animation is used more and more in the new Design System, I think that is time to remove this redundancy and must be solved by a custom component such as `PressableAnimatied` or `ButtonPressable` where all this bunch of code can be wrapped and, doing an example, this component will become a 5-6 rows component such as the following:

```typescript
export const ListItemIDP = ({
  name,
  localLogo,
  logo,
  saved,
  onPress,
  testID
}: ListItemIDP) => {
 
  const urlLogoIDP = localLogo
    ? localLogo
    : {
        uri: `${logo}${androidIdpLogoForcedRefreshed()}`
      };

  return (
    <PressableAnimated
      onPress={onPress}
      accessibilityLabel={name}
      testID={testID}
      saved={saved}
    >
      <Text style={styles.idpName}>{name}</Text>
      <Image source={urlLogoIDP} style={styles.idpLogo} />
    </Pressable>
  );
};
``` 

What do you think?

Reply:
I agree with @forrest57's suggestion, but there are still some drawbacks:
- His PR with the new hook is still under review and not merged yet.
- We should find a good abstraction that can be applied to all components, because each component has some differences between each other (e.g.: different padding, different styles, etc...). For example, the `useListItemSpringAnimation` hook suggested by @forrest57 seems to be a very good compromise between repetition and optimization, but `PressableAnimated` is not, because there's a specific style condition valid only for the `ListItemIDP` component.

Reply:
I think that it is not a problem if you have a specific style for different usages, I mean that with a `PressableAnimated` component, you could add a `style` prop that can be used into the wrapper `Animated.View` inside the `PressableAnimated` component and in this case is more scalable, for example in this case since the `saved` prop is a boolean condition used to add or not a component specific style, it become as follow:
```tsx
export const ListItemIDP = ({
  name,
  localLogo,
  logo,
  saved,
  onPress,
  testID
}: ListItemIDP) => {
 
  const urlLogoIDP = localLogo
    ? localLogo
    : {
        uri: `${logo}${androidIdpLogoForcedRefreshed()}`
      };

  return (
    <PressableAnimated
      onPress={onPress}
      accessibilityLabel={name}
      testID={testID}
      style={
        // Here you can add whatever custom component specific style you want
        saved && { paddingVertical: IOListItemIDPSavedVSpacing }
      }
    >
      <Text style={styles.idpName}>{name}</Text>
      <Image source={urlLogoIDP} style={styles.idpLogo} />
    </Pressable>
  );
};
```

Reply:
Anyway I think that I can proceed to approve this PR because it is good for me except the redundancy and I suggest to open a new task/story in backlog about the creation of a wrapper component used for the most used animations such as Pressable in order to open another PR with the development of what has been said

What do you think about?

Reply:
I feel the same way. The eventual optimization could be introduced in an ad hoc PR.